### PR TITLE
SPA authentication and authorization doc

### LINKED ISSUE #29363: .NET 8: MapIdentityApi<TUser>()
See [`MapIdentityApi<TUser>() `](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-4/#identity-api-endpoints)

Related: #30066

The main goal of the MapIdentityApi is to make it easy for developers to use ASP.NET Core Identity for authentication in JavaScript-based single page apps (SPA) or **_Blazor apps_**. 

~Since it's for Blazor app, assigning to @guardrex. If you don't want this, let me know.~

---
[Associated WorkItem - 159474](https://dev.azure.com/msft-skilling/Content/_workitems/edit/159474)

### LINKED ISSUE #30066: .NET 8: Support for tokens in lieu of cookies for Identity sign on
[Identity](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-7/#identity)

Related to #29363

---
[Associated WorkItem - 159475](https://dev.azure.com/msft-skilling/Content/_workitems/edit/159475)

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #30066 
Fixes #29363 

The diff is not useful, the .NET 8 content is in effect a new doc.

Use this [internal review link](https://review.learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?view=aspnetcore-8.0&branch=pr-en-us-31010) for the new .NET 8 content.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/security/authentication/identity-api-authorization.md](https://github.com/dotnet/AspNetCore.Docs/blob/f8165e882d33bc6438fef28d64b26b82edc39cec/aspnetcore/security/authentication/identity-api-authorization.md) | [Authentication and authorization for SPAs](https://review.learn.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?branch=pr-en-us-31010) |

### CODE REVIEW

**Review by JeremyLikness — CHANGES_REQUESTED** [review]
Few changes. @tdykstra I will also have a sample for tokens with comments right after lunch so we can complete that section.

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 1]
```diff
@@ -1,429 +1,284 @@
...
-builder.Services.AddAuthentication()
-    .AddIdentityServerJwt();
-
-builder.Services.Configure<JwtBearerOptions>(
-    IdentityServerJwtConstants.IdentityServerJwtBearerScheme,
-    options =>
-    {
-        ...
-    });
+builder.Services.AddIdentityCore<MyUser>()
```
Comment (guardrex) [inline aspnetcore/security/authentication/identity-api-authorization.md:117]:
```suggestion
builder.Services.AddIdentityCore<AppUser>()
```

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 2]
```diff
@@ -1,429 +1,284 @@
...
+Install these packages by using the [NuGet package manager in Visual Studio](/nuget/consume-packages/install-use-packages-visual-studio) or the [dotnet add package](/dotnet/core/tools/dotnet-add-package) CLI command.
 
-[!code-csharp[](~/security/authentication/identity-api-authorization/6samples/Program.cs)]
+## Create an `IdentityUser`
 
-The preceding code configures:
+Identity allows you to customize both the user information and the user database. To get started with a basic example, just use the default user information and database.
 
-* Identity with the default UI:
+Add to the project a class named `MyUser` that inherits from <xref:Microsoft.AspNetCore.Identity.IdentityUser>:
```
Comment (guardrex) [inline aspnetcore/security/authentication/identity-api-authorization.md:49]:
```suggestion
Add to the project a class named `AppUser` that inherits from <xref:Microsoft.AspNetCore.Identity.IdentityUser>:
```

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 3]
```diff
@@ -1,429 +1,284 @@
...
-
-  ```csharp
-  builder.Services.AddIdentityServer()
-      .AddApiAuthorization<ApplicationUser, ApplicationDbContext>();
-  ```
+```csharp
+using Microsoft.AspNetCore.Identity;
 
-* Authentication with an additional `AddIdentityServerJwt` helper method that configures the app to validate JWT tokens produced by IdentityServer:
+public class MyUser : IdentityUser
```
Comment (guardrex) [inline aspnetcore/security/authentication/identity-api-authorization.md:54]:
```suggestion
public class AppUser : IdentityUser
```

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 4]
```diff
@@ -1,429 +1,284 @@
...
-    ```csharp
-    app.UseAuthentication();
-    ```
+```csharp
+using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore;
 
-* The IdentityServer middleware that exposes the OpenID Connect endpoints:
+class AppDbContext(DbContextOptions<AppDbContext> options) :
+    IdentityDbContext<MyUser>(options)
```
Comment (guardrex) [inline aspnetcore/security/authentication/identity-api-authorization.md:68]:
```suggestion
    IdentityDbContext<AppUser>(options)
```

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 5]
```diff
@@ -1,429 +1,284 @@
...
-    {
-        var onTokenValidated = options.Events.OnTokenValidated;       
-        
-        options.Events.OnTokenValidated = async context =>
-        {
-            await onTokenValidated(context);
-            ...
-        }
-    });
+app.MapIdentityApi<MyUser>();
```
Comment (guardrex) [inline aspnetcore/security/authentication/identity-api-authorization.md:127]:
```suggestion
app.MapIdentityApi<AppUser>();
```

**Review by guardrex — COMMENTED** [review]
Passing recommendation to go with `AppUser`. I thought there was a time that we were trying to drop "`My`"-named things, but that just might be my imagination😄.

**Comment by tdykstra** [comment]
@guardrex I'm following the pattern in the simplified code supplied by Jeremy that uses IdentityUser directly.

File: aspnetcore/security/authentication/identity-api-authorization.md [hunk 6]
```diff
@@ -1,428 +1,251 @@
...
-               "http://www.example.com/authentication/logout-callback"));
-
-    options.ApiResources.AddApiResource("MyExternalApi", resource =>
-        resource.WithScopes("a", "b", "c"));
-});
-```
+## See also
 
-## Additional resources
+For more information, see the following resources:
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/identity-api-authorization.md:239]: `## Additional resources` is shorter.

**Review by Rick-Anderson — APPROVED** [review]

**Review by JeremyLikness — APPROVED** [review]
Looks good to me.

**Comment by guardrex** [comment]
@tdykstra ... Very good. I currently left it the original way that @JeremyLikness had it for the Blazor sample, but I think that's a good approach. I'll make a long-range tracking note to update that sample, but @JeremyLikness can let me know if switching over to `IdentityUser` for that sample app isn't the best choice for some reason. I won't get to that for a little while. I'll probably be swamped with higher priority work all the way to the ⛄ EOY holidays.

**Comment by JeremyLikness** [comment]
Using `IdentityUser` is preferred unless we're showing how to customize the user.

**Pull request merged**